### PR TITLE
Improve scribe throttling logic to only check queue size of the stream involved

### PR DESCRIPTION
**Pull request by rajubairishetti**

### CODE REVIEW

File: src/scribe_server.cpp [hunk 1]
```diff
@@ -389,33 +389,45 @@ bool scribeHandler::throttleRequest(const vector<LogEntry>&  messages) {
...
 
+  string category;
+  for (vector<LogEntry>::const_iterator msg_iter = messages.begin();
+       msg_iter != messages.end(); ++msg_iter) {
+    if ((*msg_iter).category.empty()) {
+      continue;
+    }
+    category = (*msg_iter).category;
+    break;
+  }
```
Comment (satish-mittal) [inline src/scribe_server.cpp:411]: What if category is still null after the loop? You need to handle it by not throttling in that case.

File: src/scribe_server.cpp [hunk 2]
```diff
@@ -389,33 +389,45 @@ bool scribeHandler::throttleRequest(const vector<LogEntry>&  messages) {
...
-        unsigned long long size = (*store_iter)->getSize();
-        if (size > maxQueueSize) {
-          LOG_OPER("throttle denying request for queue size <%llu>. It would exceed max queue size <%llu>", size, maxQueueSize);
-          incCounter((*store_iter)->getCategoryHandled(), "denied for queue size");
-          return true;
-        }
+  // Also note that we always check the category which is present in this request.
+  // This is a simplification based on the assumption that Log() calls does not
+  // contains messages from multiple categories(i.e. single request does not
+  // contain messages from multiple categories)
```
Comment (satish-mittal) [inline src/scribe_server.cpp]: Also can you move this comment snippet up in the beginning, and indent it properly?

File: src/scribe_server.cpp (hunk 1 above)
Comment (rajubairishetti) [inline src/scribe_server.cpp:411]: Assumption here is all messages in the request belongs to single category.

**Comment by satish-mittal** [comment]
+1

**Pull request merged**